helix-term/src/health.rs

```diff
@@ -0,0 +1,112 @@
+use crossterm::style::Stylize;
+use helix_core::{
+    config::{default_syntax_loader, user_syntax_loader},
+    syntax::load_runtime_file,
+};
+
+/// Display general diagnostics.
+pub fn general() {
+    let config_file = helix_core::config_file();
+    let lang_file = helix_core::lang_config_file();
+    let log_file = helix_core::log_file();
+    let rt_dir = helix_core::runtime_dir();
+
+    if config_file.exists() {
+        println!("Config file: {}", config_file.display());
+    } else {
+        println!("Config file: default")
+    }
+    if lang_file.exists() {
+        println!("Language file: {}", lang_file.display());
+    } else {
+        println!("Language file: default")
+    }
+    println!("Log file: {}", log_file.display());
+    println!("Runtime directory: {}", rt_dir.display());
+
+    if let Ok(path) = std::fs::read_link(&rt_dir) {
+        let msg = format!("Runtime directory is symlinked to {}", path.display());
+        println!("{}", msg.yellow());
+    }
+    if !rt_dir.exists() {
+        println!("{}", "Runtime directory does not exist.".red());
+    }
+    if rt_dir.read_dir().ok().map(|it| it.count()) == Some(0) {
+        println!("{}", "Runtime directory is empty.".red());
+    }
+}
+
+/// Display diagnostics pertaining to a particular language (LSP,
+/// highlight queries, etc).
+pub fn language(lang_str: String) {
+    let syn_loader_conf = user_syntax_loader().unwrap_or_else(|err| {
+        eprintln!("{}: {}", "Error parsing user language config".red(), err);
+        eprintln!("{}", "Using default language config".yellow());
+        default_syntax_loader()
+    });
+
+    let lang = match syn_loader_conf
+        .language
+        .iter()
+        .find(|l| l.language_id == lang_str)
+    {
+        Some(l) => l,
+        None => {
+            println!("{}", "Given language not found".red());
+            let suggestions: Vec<&str> = syn_loader_conf
+                .language
+                .iter()
+                .filter(|l| l.language_id.starts_with(lang_str.chars().next().unwrap()))
+                .map(|l| l.language_id.as_str())
+                .collect();
```

Comment:
I think it should be based on edit distance rather than this I think, but this should be a good start, maybe can add some comment on this.